### ISSUE TITLE
Azure Data Explorer Exporter time formatting

### ISSUE THREAD

**Issue by tmeadon** [open-telemetry/opentelemetry-collector-contrib#16546]
### Component(s)

exporter/azuredataexplorer

### Describe the issue you're reporting

Due to the use of `RFC3339` formatting in the ADX exporter, timestamps in spans, metrics, and logs do not include milliseconds or nanoseconds, despite this level of precision being present in the telemetry emitted by the SDK (.NET in my case).  This makes it impossible to sort spans in a given trace when they all complete within the same second - they appear to all happen simultaneously.

**Comment by tmeadon** [comment]
Looks like the new issue workflow failed so pinging code owners manually.  Hey @asaharan, @ag-ramachandran 👋

**Comment by ag-ramachandran** [comment]
@tmeadon looking into this. Will replicate and provide a fix for this

**Comment by ag-ramachandran** [comment]
@tmeadon, just got an internal PR for this. https://github.com/ag-ramachandran/opentelemetry-collector-contrib/pull/3583
We should be able to get this back into main as a PR later today

**Comment by ag-ramachandran** [comment]
@tmeadon Further update, internally testing the fix [here](https://github.com/ag-ramachandran/opentelemetry-collector-contrib/pull/3583) .  Will get this out by later today as a PR to the main repo
We also have some performance fixes & SDK fixes lined up for subsequent releases.

**Comment by tmeadon** [comment]
Awesome!  Thanks @ag-ramachandran 👍

What needs to happen following a merge to the main repo before I can consume these changes?

**Comment by ag-ramachandran** [comment]
> Awesome! Thanks @ag-ramachandran 👍
> 
> What needs to happen following a merge to the main repo before I can consume these changes?

That is right, we will raise the PR to the main repo later today or worst case tomorrow. Once it gets into the main branch it should be good in the next build

**Comment by tmeadon** [comment]
Thanks for looking at this so quickly @ag-ramachandran!

**Comment by ag-ramachandran** [comment]
Hello @tmeadon, Since the PR is merged the latest build should have this fix on the distribution. Would you be able to give this a try and close the issue if the fix looks fine.

**Comment by tmeadon** [comment]
Perfect!

![image](https://user-images.githubusercontent.com/39049630/208401700-87f6a8e8-8b9c-4e63-bd7b-35e223ecc6a2.png)

Thanks again @ag-ramachandran 😀
